lib/Sema/CSDiagnostics.cpp

```diff
@@ -2430,6 +2433,42 @@ bool ContextualFailure::tryTypeCoercionFixIt(
   return false;
 }
 
+bool ContextualFailure::tryProtocolConformanceFixIt(
+    InFlightDiagnostic &diagnostic) const {
+  auto innermostTyCtx = getDC()->getInnermostTypeContext();
+  if (!innermostTyCtx)
+    return false;
+
+  auto nominal = innermostTyCtx->getSelfNominalTypeDecl();
+  if (!nominal)
+    return false;
+
+  auto unwrappedToType =
+      ToType->lookThroughAllOptionalTypes()->getWithoutParens();
+  bool shouldOfferFixIt = nominal->getSelfTypeInContext()->isEqual(FromType) &&
+                          unwrappedToType->isExistentialType();
+  if (!shouldOfferFixIt)
+    return false;
+
+  diagnostic.flush();
+
+  auto endLoc = Lexer::getLocForEndOfToken(getASTContext().SourceMgr,
+                                           nominal->getNameLoc());
+  auto tok = Lexer::getTokenAtLocation(getASTContext().SourceMgr, endLoc);
+  auto conformanceDiag = emitDiagnostic(
+      nominal->getLoc(), diag::assign_protocol_conformance, unwrappedToType);
+
+  if (tok.is(tok::colon)) {
+    endLoc = Lexer::getLocForEndOfToken(getASTContext().SourceMgr, endLoc);
+    conformanceDiag.fixItInsert(endLoc,
```

Comment:
I don't think this is correct for classes. If one has a superclass, this will move it into the second position after the colon, which is invalid.

Reply:
Good spot! I have fixed it now.